boot/bootutil/src/loader.c

```diff
@@ -529,13 +529,14 @@ split_image_check(struct image_header *app_hdr,
     FIH_CALL(bootutil_img_validate, fih_rc, NULL, 0, loader_hdr, loader_fap,
              tmpbuf, BOOT_TMPBUF_SZ, NULL, 0, loader_hash);
     if (fih_not_eq(fih_rc, FIH_SUCCESS)) {
-        FIH_RET(fih_rc);
+        goto out;
     }
 
     FIH_CALL(bootutil_img_validate, fih_rc, NULL, 0, app_hdr, app_fap,
              tmpbuf, BOOT_TMPBUF_SZ, loader_hash, 32, NULL);
 
 out:
+    free(tmpbuf);
```

Comment:
This is a static. You need to have a `tmpbuf = NULL;`. Otherwise a subsequent call to this function will result in the freed buffer being used.